src/crypto/crypto.cpp

```diff
@@ -92,6 +92,11 @@ namespace crypto {
     sc_reduce32(&res);
   }
 
+  void coerce_valid_sec_key_from(const legacy16B_secret_key &legacy16B_mymonero_sec_seed, secret_key &dst__sec_seed) { // cn_fast_hash legacy16B_sec_seed in order to 'pad' it to 256 bits so it can be chopped to ec_scalar
+    cn_pad_by_fast_hash((uint8_t *)&legacy16B_mymonero_sec_seed, sizeof(crypto::legacy16B_secret_key),
```

Comment:
I like to trap potential pitfalls with future code modifcations:
```c++
static_assert(!epee::string_tools::has_padding<legacy16B_secret_key>(), "potential hash of padding data");
static_assert(!epee::string_tools::has_padding<secret_key>(), "writing to struct with extra data");
```
The latter isn't perfect, because it needs to be POD, but this gets close enough.

Reply:
Hm this does require introducing a epee::string_tools include to crypto.cpp... and granted most people who want to use crypto.cpp are going to be including epee... and while we may want to steer away from that requirement going forward, I think the way to do so would actually be to break epee up instead (so we can e.g. get a string_tools w/o importing networking). So with that in mind I'll go ahead and add these asserts.

Reply:
Oops, correction, it does not require including string_tools, as `has_padding` resides in `epee::`